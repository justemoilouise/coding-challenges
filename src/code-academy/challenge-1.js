import * as React from 'react';
import * as styles from '../styles/challenge.module.css';
import classNames from 'classnames';
import { Pill } from '../components/pill/pill';
import { getMean, getMode } from './challenge';

export const Challenge = () => {
  const [input, setInput] = React.useState([]);

  const onKeyDown = evt => {
    if (evt.code === 'Enter') {
      const currentTarget = evt.currentTarget;
      setInput(input.concat(parseInt(currentTarget.value)));
      currentTarget.value = undefined;
    }
  };

  return (
    <div className={classNames(styles.container, styles.gridContainer)}>
      <div>
        <div className={styles.flexContainer}>
          Input:&nbsp;
          <input type='number' className={styles.inputField} onKeyDown={onKeyDown} />
        </div>
        {input.length > 0 && (
          <div className={styles.flexContainer}>
            {input.map((value, index) => (
              <Pill key={index} variant='secondary'>
                {value}
              </Pill>
            ))}
          </div>
        )}
      </div>
      <div>
        Mean: <strong>{getMean(input).toFixed(2)}</strong>
        <br />
        <br />
        Mode: <strong>{getMode(input)}</strong>
      </div>
    </div>
  );
};
