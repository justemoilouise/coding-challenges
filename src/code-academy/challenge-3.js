import * as React from 'react';
import * as styles from '../styles/challenge.module.css';
import classNames from 'classnames';
import { Pill } from '../components/pill/pill';
import { getPrimeNumbers } from './challenge';

export const Challenge = () => {
  const inputRef = React.useRef(null);
  const [output, setOutput] = React.useState([]);

  const onClick = () => {
    if (!inputRef?.current) {
      return null;
    }
    
    const value = inputRef.current.value;
    const primeNumArr = getPrimeNumbers(value);

    setOutput(primeNumArr);
  };

  return (
    <div className={classNames(styles.container, styles.gridContainer)}>
      <div className={styles.flexContainer}>
        Input:&nbsp;<input type='number' ref={inputRef} className={styles.inputField} />
        <button className={styles.button} onClick={onClick}>Submit</button>
      </div>
      {output && (
        <div className={styles.flexContainer}>
          Output:
          <br />
          {output.map((value, index) => (
            <Pill key={index} variant='secondary'>{value}</Pill>
          ))}
        </div>
      )}
    </div>
  );
};
