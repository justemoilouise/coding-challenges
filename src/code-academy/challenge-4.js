import * as React from 'react';
import * as styles from '../styles/challenge.module.css';
import classNames from 'classnames';
import { Pill } from '../components/pill/pill';
import { arrayToBST } from './challenge';

export const Challenge = () => {
  const [input, setInput] = React.useState([]);
  const [output, setOutput] = React.useState();

  const onKeyDown = (evt) => {
    if (evt.code === 'Enter') {
      const currentTarget = evt.currentTarget;
      setInput(input.concat(parseInt(currentTarget.value)));
      currentTarget.value = '';
    }
  };

  const onClick = () => {
    const rootNode = arrayToBST(input, 0, input.length - 1);
    setOutput(rootNode);
  };

  return (
    <div className={classNames(styles.container, styles.gridContainer)}>
      <div>
        <div className={styles.flexContainer}>
          Input:&nbsp;<input type='number' onKeyDown={onKeyDown} className={styles.inputField} />
          <button className={styles.button} onClick={onClick}>Submit</button>
        </div>
        {input.length > 0 && (
          <div className={styles.flexContainer}>
            {input.map((value, index) => (
              <Pill key={index} variant='secondary'>{value}</Pill>
            ))}
          </div>
        )}
      </div>
      {output && (
        <div className={styles.flexContainer}>
          Output: {output.value}
        </div>
      )}
    </div>
  );
};
