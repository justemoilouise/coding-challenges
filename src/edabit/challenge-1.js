import * as React from 'react';
import * as styles from '../styles/challenge.module.css';
import { Pill } from '../components/pill/pill';
import { convertBinaryTextToNumber } from './challenge';

export const Challenge = () => {
  const inputRef = React.useRef();

  const [data, setData] = React.useState();

  const handleClick = () => {
    if (!inputRef?.current) {
      return null;
    }

    const input = inputRef.current.value.toLowerCase().split(' ').filter(x => ['zero', 'one'].includes(x));
    const output = convertBinaryTextToNumber(input);

    setData({
      input,
      output,
    });
    inputRef.current.value = '';
  };

  return (
    <div className={styles.container}>
      <div className={styles.flexContainer}>
        Input:&nbsp;<input ref={inputRef} className={styles.inputField} />
        <button className={styles.button} onClick={handleClick}>Convert</button>
      </div>
      {data && (
        <>
          <div className={styles.flexContainer}>
            Given:&nbsp;
            {data.input.map((num, i) => (<Pill key={i} variant='secondary'>{num}</Pill>))}
          </div>
          <div className={styles.flexContainer}>
            Output:&nbsp;<strong>{data.output}</strong>
          </div>
        </>
      )}
    </div>
  );
};
