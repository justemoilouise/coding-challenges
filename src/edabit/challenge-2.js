import * as React from 'react';
import * as styles from '../styles/challenge.module.css';
import classNames from 'classnames';
import { Pill } from '../components/pill/pill';
import { getFibonacci } from './challenge';

export const Challenge = () => {
  const inputRef = React.useRef();

  const [seq, setSeq] = React.useState();

  const handleClick = () => {
    if (!inputRef?.current) {
      return null;
    }

    setSeq(undefined);

    const input = parseInt(inputRef.current.value);
    if (input) {
      setSeq(getFibonacci(input));
    }
  };

  return (
    <div className={classNames(styles.container, styles.gridContainer)}>
      <div className={styles.flexContainer}>
        Input:&nbsp;<input type='number' min={0} ref={inputRef} className={styles.inputField} />
        <button className={styles.button} onClick={handleClick}>Submit</button>
      </div>
      {seq && (
        <div>
          Fibonacci Sequence:&nbsp;
          {seq.map((num, i) => (<Pill key={i} variant='secondary'>{num}</Pill>))}
        </div>
      )}
    </div>
  );
};
