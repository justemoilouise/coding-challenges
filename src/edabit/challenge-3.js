import * as React from 'react';
import * as styles from '../styles/challenge.module.css';
import classNames from 'classnames';
import { Pill } from '../components/pill/pill';
import { boomSeven } from './challenge';

export const Challenge = () => {
  const [input, setInput] = React.useState([]);
  const [output, setOutput] = React.useState([]);

  const onKeyDown = evt => {
    if (evt.code === 'Enter') {
      const currentTarget = evt.currentTarget;
      setInput(input.concat(parseInt(currentTarget.value)));
      currentTarget.value = '';
    }
  };

  const onClick = () => {
    setOutput(boomSeven(input));
  };

  return (
    <div className={classNames(styles.container, styles.gridContainer)}>
      <div>
        <div className={styles.flexContainer}>
          Input:&nbsp;
          <input type='number' className={styles.inputField} onKeyDown={onKeyDown} />
          <button className={styles.button} onClick={onClick}>
            Submit
          </button>
        </div>
        {input.length > 0 && (
          <div className={styles.flexContainer}>
            {input.map((value, index) => (
              <Pill key={index} variant='secondary'>
                {value}
              </Pill>
            ))}
          </div>
        )}
      </div>
      {output && (
        <div>
          <div className={styles.flexContainer}>Output:</div>
          {output.length > 0 && (
            <div className={styles.flexContainer}>
              <strong>{output}</strong>
            </div>
          )}
        </div>
      )}
    </div>
  );
};
