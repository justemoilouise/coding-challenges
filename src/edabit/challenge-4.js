import * as React from 'react';
import * as styles from '../styles/challenge.module.css';
import classNames from 'classnames';
import { towerOfHanoi } from './challenge';

export const Challenge = () => {
  const inputRef = React.useRef();

  const [output, setOutput] = React.useState([]);

  const onClick = () => {
    const value = inputRef.current.value;
    if (value > 0) {
      setOutput(towerOfHanoi(inputRef.current.value));
    }
  };

  return (
    <div className={classNames(styles.container, styles.gridContainer)}>
      <div className={styles.flexContainer}>
        Input:&nbsp;
        <input type='number' min={1} ref={inputRef} className={styles.inputField} />
        <button className={styles.button} onClick={onClick}>
          Play
        </button>
      </div>
      {output && output.length > 0 && (
        <div>
          <div className={styles.flexContainer}>Output:</div>
          <ul>
            {output.map((o, i) => (
              <li key={i}>{o}</li>
            ))}
          </ul>
        </div>
      )}
    </div>
  );
};
