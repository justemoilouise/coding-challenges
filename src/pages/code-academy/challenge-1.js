import * as React from 'react';
import { Challenge } from '../../code-academy/challenge-1';
import { CodeAcademyPageLayout } from '../../layouts/codeAcademy-pageLayout';
import data from '../../content/code-academy.json';

const breadcumbItems = [
  {
    link: '/code-academy',
    label: 'Code Academy',
  },
  {
    label: 'Challenge 1',
  },
];

const details = data[0];

const Page = () => {
  return (
    <CodeAcademyPageLayout {...details} title={`Challenge 1 - ${details.title}`} breadcumbItems={breadcumbItems}>
      <Challenge />
    </CodeAcademyPageLayout>
  );
};

export default Page;

export const Head = () => <title>Coding Challenges - Code Academy - Challenge 1</title>;
