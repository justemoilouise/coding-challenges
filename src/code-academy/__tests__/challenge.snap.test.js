import React from 'react';
import renderer from 'react-test-renderer';
import { Challenge as Challenge1 } from '../challenge-1';
import { Challenge as Challenge2 } from '../challenge-2';
import { Challenge as Challenge3 } from '../challenge-3';
import { Challenge as Challenge4 } from '../challenge-4';

describe('Code Academy challenges snapshots', () => {
  it('Challenge 1', () => {
    const tree = renderer
      .create(<Challenge1 />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Challenge 2', () => {
    const tree = renderer
      .create(<Challenge2 />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Challenge 3', () => {
    const tree = renderer
      .create(<Challenge3 />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Challenge 4', () => {
    const tree = renderer
      .create(<Challenge4 />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
