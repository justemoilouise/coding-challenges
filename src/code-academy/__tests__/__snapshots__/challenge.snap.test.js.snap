// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code Academy challenges snapshots Challenge 1 1`] = `
<div
  className=""
>
  <div>
    <div>
      Input: 
      <input
        onKeyDown={[Function]}
        type="number"
      />
    </div>
  </div>
  <div>
    Mean: 
    <strong>
      0.00
    </strong>
    <br />
    <br />
    Mode: 
    <strong>
      0
    </strong>
  </div>
</div>
`;

exports[`Code Academy challenges snapshots Challenge 2 1`] = `
<div
  className=""
>
  <div>
    <div>
      Input: 
      <input
        onKeyDown={[Function]}
        type="number"
      />
      <button
        onClick={[Function]}
      >
        Submit
      </button>
    </div>
  </div>
  <div>
    <div>
      Output:
    </div>
  </div>
</div>
`;

exports[`Code Academy challenges snapshots Challenge 3 1`] = `
<div
  className=""
>
  <div>
    Input: 
    <input
      type="number"
    />
    <button
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div>
    Output:
    <br />
  </div>
</div>
`;

exports[`Code Academy challenges snapshots Challenge 4 1`] = `
<div
  className=""
>
  <div>
    <div>
      Input: 
      <input
        onKeyDown={[Function]}
        type="number"
      />
      <button
        onClick={[Function]}
      >
        Submit
      </button>
    </div>
  </div>
</div>
`;

exports[`Code Academy challenges snapshots Challenge 5 1`] = `
<div
  className=""
>
  <div>
    Input: 
    <input
      type="number"
    />
    <button
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div>
    Output: 
  </div>
</div>
`;

exports[`Code Academy challenges snapshots Challenge 6 1`] = `
<div
  className=""
>
  <div>
    Input: 
    <input />
    <button
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div>
    Output: 
  </div>
</div>
`;

exports[`Code Academy challenges snapshots Challenge 7 1`] = `
<div
  className=""
>
  <div>
    Input: 
    <input />
    <button
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div>
    Output: 
    <strong />
  </div>
</div>
`;

exports[`Code Academy challenges snapshots Challenge 8 1`] = `
<div
  className=""
>
  <div>
    Input: 
    <input />
    <button
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div>
    Output: 
    <strong>
      undefined
    </strong>
  </div>
</div>
`;

exports[`Code Academy challenges snapshots Challenge 9 1`] = `
<div
  className=""
>
  <div>
    Input: 
    <input />
    <button
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div>
    Output: 
  </div>
</div>
`;

exports[`Code Academy challenges snapshots Challenge 10 1`] = `
<div
  className=""
>
  <div>
    <div>
      Input: 
      <input
        onKeyDown={[Function]}
        type="number"
      />
    </div>
    <div>
      K: 
      <input
        type="number"
      />
    </div>
    <button
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
</div>
`;

exports[`Code Academy challenges snapshots Challenge 11 1`] = `
<div
  className=""
>
  <div>
    <div>
      Input: 
      <input
        onKeyDown={[Function]}
        type="number"
      />
    </div>
    <div>
      X: 
      <input
        type="number"
      />
    </div>
    <button
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div>
    Output: 
    <strong />
  </div>
</div>
`;

exports[`Code Academy challenges snapshots Challenge 12 1`] = `
<div
  className=""
>
  <div>
    Input: 
    <input />
    <button
      onClick={[Function]}
    >
      Reverse
    </button>
  </div>
  <div>
    Output: 
    <strong />
  </div>
</div>
`;

exports[`Code Academy challenges snapshots Challenge 13 1`] = `
<div
  className=""
>
  <div>
    <div>
      Input: 
      <input
        onKeyDown={[Function]}
        type="number"
      />
      <button
        onClick={[Function]}
      >
        Submit
      </button>
    </div>
  </div>
  <div>
    <div>
      Output
    </div>
  </div>
</div>
`;

exports[`Code Academy challenges snapshots Challenge 14 1`] = `
<div
  className=""
>
  <div>
    <div>
      Input: 
      <input
        onKeyDown={[Function]}
        type="number"
      />
    </div>
    <div>
      K: 
      <input
        type="number"
      />
    </div>
    <button
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div>
    Output: 
    <strong />
  </div>
</div>
`;

exports[`Code Academy challenges snapshots Challenge 15 1`] = `
<div
  className=""
>
  <div>
    <div>
      Input: 
      <input
        type="number"
      />
    </div>
    <button
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div>
    <div>
      Output:
    </div>
  </div>
</div>
`;

exports[`Code Academy challenges snapshots Challenge 16 1`] = `
<div
  className=""
>
  <div>
    <div>
      Input: 
      <input
        type="number"
      />
    </div>
    <button
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div>
    Output: 
    <strong />
  </div>
</div>
`;

exports[`Code Academy challenges snapshots Challenge 17 1`] = `
<div
  className=""
>
  <div>
    <div>
      Coins: 
      <input
        onKeyDown={[Function]}
        type="number"
      />
    </div>
    <div>
      Money: 
      <input
        type="number"
      />
    </div>
    <button
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div>
    Output:
  </div>
</div>
`;

exports[`Code Academy challenges snapshots Challenge 18 1`] = `
<div
  className=""
>
  <div>
    <div>
      Input: 
      <input />
    </div>
    <div>
      Elements: 
      <input />
    </div>
    <button
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div>
    <div>
      Output:
    </div>
    <div />
  </div>
</div>
`;
